specification/communication/Communication.Messages/models.tsp

```diff
@@ -0,0 +1,322 @@
+  @doc("Message notification type is text.")
+  kind: "text";
+
+  @doc("Message content.")
+  content: string;
+}
+
+@doc("A request to send a media notification.")
+model MediaNotificationContent extends NotificationContent {
+  @doc("Message notification type is image.")
+  kind: "image";
+
+  @doc("Optional text content.")
+  content?: string;
+
+  @doc("""
+A media url for the file. Required if the type is one of the supported media
+types, e.g. image
+""")
+  mediaUri: url;
+}
+
+@doc("A request to send a template notification.")
+model TemplateNotificationContent extends NotificationContent {
+  @doc("Message notification type is template.")
+  kind: "template";
+
+  @doc("The template object used to create templates.")
+  template: MessageTemplate;
+}
+
+@doc("The template object used to create templates.")
+model MessageTemplate {
+  @doc("Name of the template.")
+  name: string;
+
+  @doc("The codes for the supported languages for templates.")
+  language: string;
+
+  @doc("The template values.")
+  values?: MessageTemplateValue[];
+
+  @doc("The binding object to link values to the template specific locations")
+  bindings?: MessageTemplateBindings;
+}
+
+@discriminator("kind")
+@doc("The class describes a parameter of a template.")
+model MessageTemplateValue {
+  @doc("Name of the Template value")
+  name: string;
+}
+
+@doc("The message template's text value information.")
+model MessageTemplateText extends MessageTemplateValue {
+  @doc("Message parameter type is text.")
+  kind: "text";
+
+  @doc("The text value.")
+  text: string;
+}
+
+@doc("""
+The message template's media value information.
+Could be an image, document or
+video.
+""")
+model MessageTemplateMedia {
+  @doc("The (public) URL of the media.")
+  url: url;
+
+  @doc("The [optional] caption of the media object.")
+  caption?: string;
+
+  @doc("The [optional] filename of the media file.")
+  fileName?: string;
+}
+
+@doc("The message template's image value information.")
+
+model MessageTemplateImage extends MessageTemplateValue {
+  @doc("Message parameter type is image.")
+  kind: "image",
+  ...MessageTemplateMedia
+}
+
+@doc("The message template's document value information.")
+
+model MessageTemplateDocument extends MessageTemplateValue {
+  @doc("Message parameter type is document.")
+  kind: "document",
+  ...MessageTemplateMedia
+}
+
+@doc("The message template's video value information.")
+
+model MessageTemplateVideo extends MessageTemplateValue {
+  @doc("Message parameter type is video.")
+  kind: "video",
+  ...MessageTemplateMedia
+}
+
+@doc("The message template's location value information.")
+model MessageTemplateLocation extends MessageTemplateValue {
+  @doc("Message parameter type is location.")
+  kind: "location";
+
+  @doc("The [Optional] name of the location.")
+  locationName?: string;
+
+  @doc("The [Optional] address of the location.")
+  address?: string;
+
+  @doc("The latitude of the location.")
+  latitude?: float64;
```

Comment:
The other two prop docs mention they're optional, so should latitude and longitude be required?

Reply:
made required